app/perfdwarfdiecache.cpp

```diff
@@ -240,6 +240,16 @@ QByteArray demangle(const QByteArray &mangledName)
             if (status == 0)
                 return demangleBuffer = dsymname;
         }
+
+#ifdef HAVE_D_DEMANGLE
+        if (mangledName[0] == '_' && mangledName[1] == 'D') {
+            extern const char* ddemangle(const char*, char*, size_t*);
```

Comment:
> still, please put this line on top similar to the rustc_demangle.h header include

Ok, but I thought this was more self-contained.

> also please expand your commit message and talk about where to find this library

The library is the file ddemangle.d, it's just a wrapper function no need to use a repo for it.

> are you OK with me doing that for you then?

Sure, I tried to do most of the integration code for you.

Reply:
where is ddemangle.d? how can I build it?

Reply:
In the Hotspot patch. You just need to run `scripts/d-demangle/package_d_demangle.sh`, it will bootstrap the library from a docker image automatically.
Then `scripts/build_appimage_in_docker.sh` should detect it just like the rust demangler.

Reply:
we need to push this into this repo so we can upstream it to perfparser

obviously the docker scripting is hotspot specific, but it should be easily possible to build perfparser with d-demangling one way or another

Reply:
>but it should be easily possible to build perfparser with d-demangling one way or another

Of course, in fact all the code should be here in perfparser.
But currently you are building perfparser from hotspot (`hotspot/3rdparty/perfparser.cmake`) including the rust demangler, my patch is based on this approach.
The logical procedure would be to apply this now as is and then move code related to perfparser to its repo.

Reply:
Well, the rust demangler is a bit different as it's an external dependency and not build together with perfparser. Please take a look at the upstreamed patch (which may be a bit different to the current status in the hotspot branch of perfparser):

https://codereview.qt-project.org/c/qt-creator/perfparser/+/300698

Reply:
I see.
A CMake script seems the solution, something that can find D in the system to compile `ddemangle.d` if `WITH_D_DEMANGLE` is enabled.